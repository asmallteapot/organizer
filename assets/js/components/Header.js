import React from 'react'
import { Link } from 'react-router-dom'
import { Paper } from 'material-ui'

export default (props) => {
  if (props.title) {
    return (
      <header className="row">
      <Paper className="small-10 columns mastiff">
        <h1>{props.title}</h1>
      </Paper>
      <div className="small-2 columns logo">
        <Link to="/">East Bay Forward</Link>
      </div>
    </header>
    )
  } else {
    return (
      <header className="row expanded">
        <div className="small-12 columns logo logo-full">
          <Link to="/">East Bay Forward</Link>
        </div>
      </header>
    )
  }
}
