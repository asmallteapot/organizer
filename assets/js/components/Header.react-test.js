import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter as Router }  from 'react-router-dom'
import Header from './Header'
import ThemeProvider from 'material-ui/styles/MuiThemeProvider';

const WrappedHeader = (props) => (
  <ThemeProvider>
    <Router>
      <Header {...props} />
    </Router>
  </ThemeProvider>
)

test('Smoke test', () => {
  const component = mount(<WrappedHeader />);

  expect(component.html()).toMatchSnapshot();
})

test('Setting titles', () => {
  const component = mount(<WrappedHeader />);

  const blankHtml = component.html();
  expect(blankHtml).toMatchSnapshot();

  component.setProps({title: 'title'});
  expect(component.html()).toMatchSnapshot();

  component.setProps({subtitle: 'subtitle'});
  expect(component.html()).toMatchSnapshot();

  component.setProps({title: undefined, subtitle: undefined});
  expect(component.html()).toMatch(blankHtml);
})
