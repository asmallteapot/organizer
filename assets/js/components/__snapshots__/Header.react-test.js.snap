// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting titles 1`] = `"<header class=\\"row expanded\\"><div class=\\"small-12 columns logo logo-full\\"><a href=\\"/\\">East Bay Forward</a></div></header>"`;

exports[`Setting titles 2`] = `
"<header class=\\"row\\"><div class=\\"small-10 columns mastiff\\" style=\\"color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px;\\"><h1>title</h1></div><div class=\\"small-2 columns logo\\"><a href=\\"/\\">East Bay Forward</a></div></header>"
`;

exports[`Setting titles 3`] = `
"<header class=\\"row\\"><div class=\\"small-10 columns mastiff\\" style=\\"color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px;\\"><h1>title</h1></div><div class=\\"small-2 columns logo\\"><a href=\\"/\\">East Bay Forward</a></div></header>"
`;

exports[`Smoke test 1`] = `"<header class=\\"row expanded\\"><div class=\\"small-12 columns logo logo-full\\"><a href=\\"/\\">East Bay Forward</a></div></header>"`;
